lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -546,7 +546,11 @@ public function getAlterTableSQL(TableDiff $diff)
                 $columnDefinition['autoincrement'] = false;
 
                 // here was a server version check before, but DBAL API does not support this anymore.
-                $query = 'ALTER ' . $oldColumnName . ' TYPE ' . $type->getSQLDeclaration($columnDefinition, $this);
+                $customUsing = '';
+                if ($columnDiff->hasChanged('using')) {
```

Comment:
``` php
$query = 'ALTER ' . $oldColumnName . ' TYPE ' . $type->getSQLDeclaration($columnDefinition, $this);
if ($columnDiff->hasChanged('using')) {
    $query = $query . ' USING ' . $column->getCustomSchemaOption('using');
}
```

This way, `$customUsing` shouldn't be needed.

Reply:
The way it reads right now is weird: the `using` property of the column has changed. I'd rather introduce a new pair of methods for setting and retrieving this expression of a very specific type `string` than put it into an `array<string, mixed>`.

We also need a usage example. Not sure how it's supposed to be used outside of DBAL.

Reply:
I've done the code changes above.

Reply:
@morozov @shemgp Sure, new methods for `using` is also possible. If it makes sense, I would like to see this for other RDBMS too.

Reply:
@morozov, @SenseException,

1. Mysql/MariaDB doesn't have USING construction:
https://mariadb.com/kb/en/library/alter-table/

2. Oracle has method USING:
https://docs.oracle.com/cd/B28359_01/server.111/b28286/statements_3001.htm#i2103924

[ USING 'encrypt_algorithm' ]
[ IDENTIFIED BY password ]
[ [ NO ] SALT ]

But I'have doesn't work with Oracle.

Sorry.


3. SQL Server doesn't have USING method:
https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-table-transact-sql?view=sql-server-2017

I think this is a rather specific thing and in this context is not required.

This Merge Request is also needed for me in solving my Postgres SQL tasks, I have never worked with Oracle and I don’t know the functionality of the USING analogue and, all the more, I cannot write tests.

I think it will be right to give you @todo and implement it for other languages later so that the author of this commit and I can continue to developing now.

Reply:
Hi any update on this? I think this one will game changer to support across multiple databases. Just a thought, it'd be good if we can just implement the specific db migration using `trait` inside the migration class to indicate specific database migration / even a version.  (e.g. trait `use PostgresOnly`  etc.)